### PR TITLE
Add/integration tests

### PR DESCRIPTION
**Pull request by jakobe**
Added test utilities for checking actual rendered style through `window.getComputedStyle(element).getPropertyValue(propertyName)`.
Can check any css property for any string value:
`expect(element).toHaveStyle({ 'border-radius': '16px' });`

Can also check for specific Kirby Design Tokens, like color, size and font-size using similar api as SCSS implementation (e.g. `height: size('xl')`).

Color:
```
it('should render with correct background-color', () => {
    expect(element).toHaveThemeColorStyle({
      'background-color': getColor('primary'),
    });
  });

it('should render with correct color', () => {
    expect(element).toHaveThemeColorStyle({
      color: getColor('primary', 'contrast')
    });
  });
```

Size:
```
it('should render with correct height', () => {
    expect(element).toHaveStyle({ height: size('xl') });
  });
```

Font-Size:
```
it('should render with correct font-size', () => {
    expect(element).toHaveStyle({ 'font-size': fontSize('s') });
  });
```

### CODE REVIEW

**Comment by jakobe** [comment]
> LGTM, however the custom matcher .toHaveStyle is that needed? I can see that spectator already has a matcher which looks like doing the same thing:
> image

Yes it does, but it only checks the inline style attributes for the given keys, where as I check the rendered style

File: src/kirby/components/button/button.component.integration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+      PageActionsComponent,
+      PageActionsDirective,
+      PageTitleDirective,
+      PageToolbarTitleDirective,
+      FitHeadingDirective,
+    ],
+  });
+
+  beforeEach(() => {
+    jasmine.addMatchers(ElementCssCustomMatchers);
```
Comment (askarby) [inline src/kirby/components/button/button.component.integration.spec.ts]: Consider registering the matchers globally

File: src/kirby/components/button/button.component.integration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+describe('ButtonComponent with size directive', () => {
+  let spectator: SpectatorHost<ButtonComponent>;
+  let element: HTMLButtonElement;
+  const createHost = createHostFactory({
+    component: ButtonComponent,
+    declarations: [ButtonComponent, SizeDirective, MockComponent(IconComponent)],
+  });
+
+  beforeEach(() => {
+    jasmine.addMatchers(ElementCssCustomMatchers);
```
Comment (askarby) [inline src/kirby/components/button/button.component.integration.spec.ts]: Consider registering the matchers globally

File: src/kirby/components/button/button.component.spec.ts [hunk 3]
```diff
@@ -1,28 +1,276 @@
...
+  const createHost = createHostFactory({
+    component: ButtonComponent,
+    declarations: [ButtonComponent, MockComponent(IconComponent)],
+  });
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(ButtonComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    jasmine.addMatchers(ElementCssCustomMatchers);
```
Comment (askarby) [inline src/kirby/components/button/button.component.spec.ts]: Consider registering the matchers globally

File: src/kirby/components/button/button.component.integration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+describe('ButtonComponent configured with icon only', () => {
+  let spectator: SpectatorHost<ButtonComponent>;
+  let element: HTMLButtonElement;
+  const createHost = createHostFactory({
+    component: ButtonComponent,
+    declarations: [ButtonComponent, SizeDirective, MockComponent(IconComponent)],
+  });
+
+  beforeEach(() => {
+    jasmine.addMatchers(ElementCssCustomMatchers);
```
Comment (askarby) [inline src/kirby/components/button/button.component.integration.spec.ts]: Consider registering the matchers globally

File: src/kirby/components/button/button.component.integration.spec.ts [hunk 5]
```diff
@@ -0,0 +1,269 @@
+/// <reference path="../../testing/element-css-custom-matchers.d.ts"/>
```
Comment (askarby) [inline src/kirby/components/button/button.component.integration.spec.ts]: Consider adding definition reference globally

File: src/kirby/components/button/button.component.spec.ts [hunk 6]
```diff
@@ -1,28 +1,276 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
+/// <reference path="../../testing/element-css-custom-matchers.d.ts"/>
```
Comment (askarby) [inline src/kirby/components/button/button.component.spec.ts]: Consider adding definition reference globally

File: src/kirby/testing/test-helper.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+export class TestHelper {
+  public static whenHydrated(node: Node, timeout: number = 2000) {
```
Comment (askarby) [inline src/kirby/testing/test-helper.ts]: When knowing the implementation that ionic components adds a "hydrated" class to shadow dom, you can understand what's going on here.... if I didn't have that knowledge... I would be thinking "what's going on here".

Consider adding a single line (or two) of TSDoc to the method?

**Pull request merged**